### ISSUE TITLE
Problem Installing Firedrake

### ISSUE THREAD

**Issue by felipeaugustogudes** [firedrakeproject/firedrake#1533]
Hi! 

I had some problems installing Firedrake. Attached is the installation log. 

I would like some help understanding what went wrong with the installation. 

I was unable to execute the proposed tests in the step by step installation.

[firedrake-install.log](https://github.com/firedrakeproject/firedrake/files/3814528/firedrake-install.log)

**Comment by wence-** [comment]
Hi @felipeaugustogudes!

It looks like the petsc configure is not safe to one of UTF-8 characters or spaces in the path name. (FWIW spaces in pathnames are often problematic for many unix-like things).

Can you try an installation in a directory without spaces in the path?

**Comment by felipeaugustogudes** [comment]
Hi! I tried to install as directed. The error below appears when compiling the tests. The instalation  log is attached.

[firedrake-install.log](https://github.com/firedrakeproject/firedrake/files/3814961/firedrake-install.log)

    Building extension modules
Traceback (most recent call last):
  File "setup.py", line 8, in <module>
    import numpy as np
ModuleNotFoundError: No module named 'numpy'
    Linting firedrake codebase
/home/paopc/Documentos/firedrake/bin/python: No module named flake8
Makefile:8: recipe for target 'lint' failed
make: *** [lint] Error 1
[firedrake-install.log](https://github.com/firedrakeproject/firedrake/files/3814945/firedrake-install.log)

**Comment by wence-** [comment]
OK, this time the installation failed for this reason:

```
    *******************************************************************************
             UNABLE to CONFIGURE with GIVEN OPTIONS    (see configure.log for details):
    -------------------------------------------------------------------------------
    Your libraries are from MPICH but it appears your mpiexec is from OpenMPI
    *******************************************************************************
```

So this looks like you need to configure your system to use MPICH both as the MPI library and the mpiexec runner.

You can either do this by doing `update-alternatives --config mpiexec` and selecting MPICH (may need sudo) as well as `update-alternatives --config mpicc` and selecting MPICH.

If you don't want to do that, you can instead tell `firedrake-install` where all your compilers and runners are and then we set them up in the virtual environment.

```
firedrake-install --mpicc mpicc.mpich --mpicxx mpicxx.mpich --mpif90 mpif90.mpich --mpiexec mpiexec.mpich
```

**Comment by felipeaugustogudes** [comment]
Hi! Again, I try to install Firedrake, but now I stayed as sudo.

I use this comand: sudo python3 firedrake-install --mpicc mpicc.mpich --mpicxx mpicxx.mpich --
mpif90 mpif90.mpich --mpiexec mpiexec.mpich

Instalation log is attached.
[firedrake-install.log](https://github.com/firedrakeproject/firedrake/files/3815431/firedrake-install.log)


Before, I try do execute the tests and this error appears:

    Building extension modules
/bin/sh: 1: cannot create build.log: Permission denied
cat: build.log: Arquivo ou diretório inexistente
Makefile:4: recipe for target 'modules' failed
make: *** [modules] Error 1

**Comment by wence-** [comment]
This install also failed:

```
             UNABLE to CONFIGURE with GIVEN OPTIONS    (see configure.log for details):
    -------------------------------------------------------------------------------
    Unable to run mpiexec with option "-n 1"
    --------------------------------------------------------------------------
    mpiexec has detected an attempt to run as root.
    Running at root is *strongly* discouraged as any mistake (e.g., in
    defining TMPDIR) or bug can result in catastrophic damage to the OS
    file system, leaving your system in an unusable state.

    You can override this protection by adding the --allow-run-as-root
    option to your cmd line. However, we reiterate our strong advice
    against doing so - please do so at your own risk.
```

I (like the PETSc configure) _strongly_ recommend that you don't run as root.

`firedrake-install` is designed to only require sudo to install package-manager dependencies (and runs that with sudo asking for your password).

**Comment by felipeaugustogudes** [comment]
Hi! I tried to install Firedrake again, but now I don't use sudo. However, the erros on instalation happens again.  I follow your help and use: python3 firedrake-install --mpicc mpicc.mpich --mpicxx mpicxx.mpich --mpif90 mpif90.mpich --mpiexec mpiexec.mpich

  File "setup.py", line 8, in <module>
    import numpy as np
ModuleNotFoundError: No module named 'numpy'
    Linting firedrake codebase
/home/paopc/Documentos/firedrake/bin/python: No module named flake8
Makefile:8: recipe for target 'lint' failed
make: *** [lint] Error 1
[firedrake-install.log](https://github.com/firedrakeproject/firedrake/files/3815872/firedrake-install.log)

**Comment by wence-** [comment]
I think this was now fixed after #1535, so I will close.
